canonical-web-and-design/assets.ubuntu.com
# docker & k8s configuration

## Done
- Dockerfile
- Konf file: need testing

## QA
- Build the Docker image: `DOCKER_BUILDKIT=1 docker build . -t prod-comms.docker-registry.canonical.com/assets.ubuntu.com:latest`
- Run a container based on the image: `docker run --network="host" --env-file=.env prod-comms.docker-registry.canonical.com/assets.ubuntu.com`
- The server should run at port 80

### Testing k8s configuration
There is no automatic script to deploy the conf file, the way I tested it:
1. Create secrets:
Here is the content of first secrets file:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: assets-ubuntu
data:
  # Base64 decoded: postgresql://assets:password@192.168.1.88:5432/assets
  database-url: cG9zdGdyZXNxbDovL2Fzc2V0czpwYXNzd29yZEAxOTIuMTY4LjEuODg6NTQzMi9hc3NldHM=
  os-auth-url: aHR0cDovL2xvY2FsaG9zdDo4MDgwL2F1dGgvdjEuMA==
  os-username: dGVzdDp0ZXN0ZXI=
  os-password: dGVzdGluZw==
```
And the second one:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: secret-keys
data:
  assets-ubuntu-com: ZGV2
```
Just add them with the command:
```bash
microk8s.kubectl apply -f your-secrets.yml
```
2. Generate the deployment script: 
```bash
snap install konf && konf demo konf/site.yaml > konf/deploy.yaml
```
3. Apply the configuration:
```bash
microk8s.kubectl create -f konf/deploy.yaml
```
That's it!
